search-app/src/cmr/search/services/association_service.clj

```diff
@@ -0,0 +1,236 @@
+(ns cmr.search.services.association-service
+  "Provides functions for associating and dissociating variables/services to collections"
```

Comment:
Tag association is keyed off tag-key rather than concept-id like variable/service association. So the code is slightly different and we have pushed off the potential refactoring until later. Added to docstring.